tests/testthat/test-shinytest2-tm_a_pca.R

```diff
@@ -0,0 +1,164 @@
+app_driver_tm_a_pca <- function() {
+  # Dataset only used once
+  data <- within(teal.data::teal_data(), {
+    require(nestcolor)
+
+    USArrests <- USArrests # nolint: object_name.
+  })
+  teal.data::datanames(data) <- "USArrests"
+
+
+  init_teal_app_driver(
+    data = data,
+    modules = tm_a_pca(
+      dat = teal.transform::data_extract_spec(
+        dataname = "USArrests",
+        select = teal.transform::select_spec(
+          choices = teal.transform::variable_choices(
+            data = data[["USArrests"]],
+            c("Murder", "Assault", "UrbanPop", "Rape")
+          ),
+          selected = c("Murder", "Assault"),
+          multiple = TRUE
+        )
+      )
+    )
+  )
+}
+
+testthat::test_that("e2e - tm_a_pca: Data selection (data_extract) changes eigenvector table", {
+  skip_if_too_deep(5)
+
+  app_driver <- app_driver_tm_a_pca()
+
+  # Data selection (adds rows to tables)
+  app_driver$set_active_module_input("dat-dataset_USArrests_singleextract-select", c("Murder", "Assault"), wait = FALSE)
+  app_driver$expect_no_validation_error()
+
+  testthat::expect_match(app_driver$get_active_module_output("tbl_eigenvector"), "Assault")
+
+  testthat::expect_failure(
+    testthat::expect_match(app_driver$get_active_module_output("tbl_eigenvector"), "UrbanPop")
+  )
```

Comment:
Maybe `expect_no_match` instead?
```suggestion
  testthat::expect_no_match(app_driver$get_active_module_output("tbl_eigenvector"), "UrbanPop")
```